controllers/AllCommentsForAnArticleController.js

```diff
@@ -0,0 +1,24 @@
+const {fetchAllCommentsForAnArticle} = require('../models/AllCommentsForAnArticleModel')
+
+const getAllCommentsForAnArticle = async(req,res,next) =>{
+
+
+    const { article_id } = req.params;
+
+    // Validate article_id
+    if (isNaN(article_id)) {
+        return res.status(400).json({ error: 'Invalid article ID' });
+    }
```

Comment:
isNaN will only determines whether a value is [NaN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN), - this isn't a check for the article ID not being a number. 

First you'd want to convert it to a number with Number(), and that would output NaN if it can't turn the value into a number. Then you can check if the result of that is NaN, which will tell you if that article ID is valid or not